web/packages/teleport/src/Console/DocumentSsh/DocumentSsh.tsx

```diff
@@ -143,7 +136,15 @@ function DocumentSsh({ doc, visible }: PropTypes) {
           <Indicator />
         </Box>
       )}
-      {mfa.requested && <AuthnDialog mfa={mfa} onCancel={closeDocument} />}
+      {mfa.mfaChallenge && <AuthnDialog {...mfa} onCancel={closeDocument} />}
+      {ft.mfaChallenge && (
+        <AuthnDialog
+          mfaChallenge={ft.mfaChallenge}
+          submitMfa={ft.submitMfa}
+          submitAttempt={ft.submitMfaAttempt}
+          onCancel={ft.clearMfaChallenge}
+        />
+      )}
```

Comment:
No, the `mfa` authndialog will always be handled and complete before we get to the `ft` authndialog. In the new updated code it would be easy to reuse the same mfa state if you're concerned about it though.